### PR TITLE
Changes for displaying FACETS data

### PR DESCRIPTION
**Pull request by averyniceday**
# What? Why?
Based on requested changes

Changes proposed in this pull request:
- add Cancer Cell Fraction as its own column
- remove tooltip displaying CCF in the clonal column

- fixed issue where Mutant Copies column was displayed even when all values are NA
- column hiding dependent on both purity (clinical attr) and total copy number (mutation attr) being available

**Patient View**
screen shot 2018-12-06 at 4 10 34 pm

**Results View**
screen shot 2018-12-06 at 4 12 06 pm

**Additional note:** waiting for feedback on renaming column `Clonal` to `Clonality` and whether display values should be `true/false` or `yes,no`  

# Checks
- [ ] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). If you are part of the
cBioPortal organization, notify the approprate team (remove inappropriate):

@cBioPortal/frontend

If you are not part of the cBioPortal organization look at who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them here:

### CODE REVIEW

**Comment by averyniceday** [comment]
@angelicaochoa updated with screenshots! There might be additional small changes depending on feedback like renaming columns

File: src/shared/components/mutationTable/column/CancerCellFractionColumnFormatter.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    public static getCcfMCopiesValue(data:Mutation[]):number {
+        const ccfMCopiesValue = data[0].ccfMCopies;
+        return ccfMCopiesValue;
+    }
+
+    public static getCancerCellFractionValue(data:Mutation[]):string {
+        let textValue:string = "";
+        const ccfMCopiesValue = CancerCellFractionColumnFormatter.getCcfMCopiesValue(data);
+        if (floatValueIsNA(ccfMCopiesValue)) {
```
Comment (sheridancbio) [inline src/shared/components/mutationTable/column/CancerCellFractionColumnFormatter.tsx]: :+1:

File: src/shared/components/mutationTable/column/CancerCellFractionColumnFormatter.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            textValue = "NA";
+        } else {
+            textValue = ccfMCopiesValue.toFixed(2);
+        }
+        return textValue;
+    }
+
+    public static getTextValue(data:number):string {
```
Comment (sheridancbio) [inline src/shared/components/mutationTable/column/CancerCellFractionColumnFormatter.tsx]: is this function called anywhere? Maybe this replaces a default function?

Comment (angelicaochoa) [reply src/shared/components/mutationTable/column/CancerCellFractionColumnFormatter.tsx]: @averyniceday 🔔 ☝️

File: src/shared/components/mutationTable/column/DiscreteCNAColumnFormatter.tsx [hunk 3]
```diff
@@ -163,13 +163,24 @@ export default class DiscreteCNAColumnFormatter {
         );
     }
 
-    public static getSortValue(data:Mutation[], molecularProfileIdToMolecularProfile: {[molecularProfileId:string]:MolecularProfile}, cache:DiscreteCNACache) {
-        return DiscreteCNAColumnFormatter.getTdValue(DiscreteCNAColumnFormatter.getData(data, molecularProfileIdToMolecularProfile, cache));
+    public static getSortValue(data:Mutation[], molecularProfileIdToMolecularProfile: {[molecularProfileId:string]:MolecularProfile}, cache:DiscreteCNACache, sampleIdToClinicalDataMap: {[key: string]:ClinicalData[]}|undefined) {
+        const facetsCNAData = DiscreteCNAColumnFormatter.getFacetsCNAData(data, sampleIdToClinicalDataMap);
+        if (facetsCNAData!= "NA") {
```
Comment (angelicaochoa) [inline src/shared/components/mutationTable/column/DiscreteCNAColumnFormatter.tsx]: space before operator

**Pull request merged**